tests/ait/core/server/test_handler.py

```diff
@@ -62,6 +67,59 @@ def test_execute_handler_returns_handle_return_on_input(self, handle_mock):
         assert returned == "SpecialReturn"
 
 
+class TestHandlerClassWith1553HSPacket(object):
+    tlm_dict = tlm.getDefaultDict()
+    pkt_data = bytearray(b"\x02\xE7\x40\x00\x00\x00\x01")
+    pkt_1553 = tlm_dict['1553_HS_Packet']
+    handler = PacketHandler(pkt_data, packet="1553_HS_Packet")
+
+    def test_word_array(self):
+        packet = tlm.Packet(self.pkt_1553, self.pkt_data)
+        assert packet.words.__len__() == 3.5
+
+    def test_execute_handler_returns_handle_return_on_input(self):
+        packet = tlm.Packet(self.pkt_1553, self.pkt_data)
+        result = self.handler.handle(packet)
+        assert 'Ethernet 1553 packet' in str(result)
+
+    # Test packet length by sending a Ethernet_HS_Packet to a 1553_HS_Packet Handler
+    def test_bad_packet_length(self):
+        ethernet_pkt = self.tlm_dict['Ethernet_HS_Packet']
+        e_packet = tlm.Packet(ethernet_pkt, self.pkt_data)
+        with pytest.raises(ValueError):
+            self.handler.handle(e_packet)
+
+    def test_packet_name_error_and_no_packet_type(self):
+        pkt_data = bytearray(b"\x02\xE7\x40\x00\x00\x00\x01")
+        with pytest.raises(ValueError):
+            PacketHandler(pkt_data, packet="1553_HS_Packe")
+        with pytest.raises(ValueError):
+            PacketHandler(pkt_data)
+
+
+class TestHandlerClassWithEthernetHSPacket(object):
+    tlm_dict = tlm.getDefaultDict()
+    pkt_data = bytearray(b"\x02\xE7\x40\x00\x00\x00\x01\x07\x08\x0a")
+    ethernet_pkt = tlm_dict['Ethernet_HS_Packet']
+    handler = PacketHandler(pkt_data, packet="Ethernet_HS_Packet")
+
+    def test_word_array(self):
+        e_packet = tlm.Packet(self.ethernet_pkt, self.pkt_data)
+        assert e_packet.words.__len__() == 5
+
+    def test_execute_handler_returns_handle_return_on_input(self):
+        e_packet = tlm.Packet(self.ethernet_pkt, self.pkt_data)
+        result = self.handler.handle(e_packet)
+        assert 'Ethernet Health and Status Packet' in str(result)
+
+    # Send a 1553 packet to an Ethernet_HS_Packet Handler
+    def test_bad_packet_length(self):
+        pkt_1553 = self.tlm_dict['1553_HS_Packet']
+        packet = tlm.Packet(pkt_1553, self.pkt_data)
+        with pytest.raises(ValueError):
+            self.handler.handle(packet)
```

Comment:
`handle()` should not be getting called with a `tlm.Packet` object as the input. It should be receiving raw bytes. The expectation on these handlers is they're classifying raw data as a particular packet type. If you already have a packet why in the world would you be doing that?

The docstring on the handle functions even specifies this

```
 """
        Params:
            input_data:   message received by stream
        Returns:
            tuple of packet UID and message received by stream
        """
 ```

Reply:
ok - I did not realize 'message received by stream' meant 'raw bytes'.